src/components/page-container/PageContainer.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useParams } from 'react-router-dom';
+
+import { LearningHeader as Header } from '@edx/frontend-component-header';
+import Footer from '@edx/frontend-component-footer';
+import { Spinner } from '@edx/paragon';
+
+import { getCohorts, getCourseHomeCourseMetadata } from './data/api';
+
+export const CourseMetadataContext = React.createContext();
+
+export default function PageContainer(props) {
+  const { children } = props;
+  const { courseId } = useParams();
+
+  const [courseMetadata, setCourseMetadata] = useState();
+
+  useEffect(() => {
+    async function fetchCourseMetadata() {
```

Comment:
Eventually, we may want to extract this as a separate function and pass it to the `PageContainer` as a prop, so that the component is more generic. This way, if a new page down the line needs more/different data, we don't have to modify this component.

For now, I'm happy with the way its done here. I don't know if we need to refactor this until that case comes up.

Reply:
Talked with TJ offline about this. I'm going to leave it as-is for now so we can continue to focus on other release of the MFE. We may need to come back around to this in the future when/if we add new functionality to the Comms MFE. For now, this shouldn't hurt staying the way it is.